libraries/heft-config-file/src/NonProjectConfigurationFile.ts

```
@@ -0,0 +1,91 @@
+  }
+
+  /**
+   * Load the configuration file at the specified absolute path, automatically resolving
+   * `extends` properties. Will throw an error if the file cannot be found.
+   */
+  public async loadConfigurationFileAsync(
+    terminal: ITerminal,
+    filePath: string
+  ): Promise<TConfigurationFile> {
+    return await this._loadConfigurationFileInnerWithCacheAsync(
+      terminal,
+      filePath,
+      new Set<string>(),
+      undefined
+    );
+  }
+
+  /**
+   * This function is identical to {@link NonProjectConfigurationFile.loadConfigurationFile}, except
+   * that it returns `undefined` instead of throwing an error if the configuration file cannot be found.
+   */
+  public tryLoadConfigurationFile(terminal: ITerminal, filePath: string): TConfigurationFile | undefined {
+    try {
+      return this.loadConfigurationFile(terminal, filePath);
+    } catch (e) {
+      if (FileSystem.isNotExistError(e as Error)) {
+        return undefined;
+      }
+      throw e;
+    }
+  }
+
+  /**
+   * This function is identical to {@link NonProjectConfigurationFile.loadConfigurationFileAsync}, except
+   * that it returns `undefined` instead of throwing an error if the configuration file cannot be found.
+   */
+  public async tryLoadConfigurationFileAsync(
+    terminal: ITerminal,
+    filePath: string
+  ): Promise<TConfigurationFile | undefined> {
+    try {
+      return await this.loadConfigurationFileAsync(terminal, filePath);
+    } catch (e) {
+      if (FileSystem.isNotExistError(e as Error)) {
+        return undefined;
+      }
+      throw e;
+    }
+  }
+
+  protected _tryLoadConfigurationFileInRig(
+    terminal: ITerminal,
+    rigConfig: IRigConfig,
+    visitedConfigurationFilePaths: Set<string>
+  ): TConfigurationFile | undefined {
+    // This is a no-op because we don't support rigging for non-project configuration files
+    return undefined;
```

Should this throw an error? Or is this handled by falling back when `undefined` is returned.

---

It may or may not be an error, depending on the calling function.